network connectivity in travis with kind

Hi, 

pods started in kind on travis seem to have lost network connectivity 

https://travis-ci.com/lukasheinrich/examples/builds/113218001

```
sudo: required
language: go
go:
    - "1.12"

services:
  - docker

install: true

script: 
  - echo "Run your tests here"

before_install:
  # Download and install Kind and kubectl
  - GO111MODULE=on go get sigs.k8s.io/kind
  - kind create cluster --config kind-config.yaml
  - export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
  - curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
  - chmod +x ./kubectl
  - curl http://google.com
  - docker run --rm -it alpine sh -c 'apk add curl;curl http://google.com'
  - ./kubectl run -it  hello --image alpine -- sh -c 'apk add curl;curl http://google.com'
```

the last command gives

```
kubectl run --generator=deployment/apps.v1 is DEPRECATED and will be removed in a future version. Use kubectl run --generator=run-pod/v1 or kubectl create instead.
If you don't see a command prompt, try pressing enter.
ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.9/main: temporary error (try again later)
WARNING: Ignoring APKINDEX.b89edf6e.tar.gz: No such file or directory
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.9/community: temporary error (try again later)
WARNING: Ignoring APKINDEX.737f7e01.tar.gz: No such file or directory
ERROR: unsatisfiable constraints:
  curl (missing):
    required by: world[curl]
sh: curl: not found
```

this seems to have been introduced with newer versions of kind (or travis) since I have successfully used this before.  Docker seems to be unaffected (see `docker run` above).



Any ideas what this might be?

If this should be moved to an issue at https://github.com/kind-ci please let me know.

I added logging to the coreDNS pods and seem I was wrong, the problem is not the one I was thinking, however the solution changing the namesever is still valid.
Let me do a few tests and check if it can be solved in kind so it's transparent to the user.

```
 for p in $(./kubectl get pods --namespace=kube-system -l k8s-app=kube-dns -o name); do ./kubectl logs --namespace=kube-system $p; done
.:53
2019-05-27T11:19:36.353Z [INFO] CoreDNS-1.3.1
2019-05-27T11:19:36.353Z [INFO] linux/amd64, go1.11.4, 6b56a9c
CoreDNS-1.3.1
linux/amd64, go1.11.4, 6b56a9c
2019-05-27T11:19:36.353Z [INFO] plugin/reload: Running configuration MD5 = 599b9eb76b8c147408aed6a0bbe0f669
2019-05-27T11:19:42.355Z [ERROR] plugin/errors: 2 6195421070260342714.8357799154935943776. HINFO: read udp 10.244.0.3:52331->169.254.169.254:53: i/o timeout
2019-05-27T11:19:45.355Z [ERROR] plugin/errors: 2 6195421070260342714.8357799154935943776. HINFO: read udp 10.244.0.3:41512->169.254.169.254:53: i/o timeout
2019-05-27T11:19:46.354Z [ERROR] plugin/errors: 2 6195421070260342714.8357799154935943776. HINFO: read udp 10.244.0.3:49238->169.254.169.254:53: i/o timeout
2019-05-27T11:19:47.354Z [ERROR] plugin/errors: 2 6195421070260342714.8357799154935943776. HINFO: read udp 10.244.0.3:57052->169.254.169.254:53: i/o timeout
2019-05-27T11:19:50.355Z [ERROR] plugin/errors: 2 6195421070260342714.8357799154935943776. HINFO: read udp 10.244.0.3:49625->169.254.169.254:53: i/o timeout
2019-05-27T11:19:53.356Z [ERROR] plugin/errors: 2 6195421070260342714.8357799154935943776. HINFO: read udp 10.244.0.3:35345->169.254.169.254:53: i/o timeout
2019-05-27T11:19:56.357Z [ERROR] plugin/errors: 2 6195421070260342714.8357799154935943776. HINFO: read udp 10.244.0.3:59890->169.254.169.254:53: i/o timeout
2019-05-27T11:19:59.358Z [ERROR] plugin/errors: 2 6195421070260342714.8357799154935943776. HINFO: read udp 10.244.0.3:52423->169.254.169.254:53: i/o timeout
2019-05-27T11:20:01.214Z [ERROR] plugin/errors: 2 www.google.com. AAAA: read udp 10.244.0.3:37466->169.254.169.254:53: i/o timeout
2019-05-27T11:20:02.358Z [ERROR] plugin/errors: 2 6195421070260342714.8357799154935943776. HINFO: read udp 10.244.0.3:40731->169.254.169.254:53: i/o timeout
2019-05-27T11:20:05.217Z [ERROR] plugin/errors: 2 www.google.com.google.internal. AAAA: read udp 10.244.0.3:47136->169.254.169.254:53: i/o timeout
2019-05-27T11:20:05.359Z [ERROR] plugin/errors: 2 6195421070260342714.8357799154935943776. HINFO: read udp 10.244.0.3:50467->169.254.169.254:53: i/o timeout
2019-05-27T11:20:07.217Z [ERROR] plugin/errors: 2 www.google.com. AAAA: read udp 10.244.0.3:35281->169.254.169.254:53: i/o timeout
2019-05-27T11:20:11.220Z [ERROR] plugin/errors: 2 www.google.com.google.internal. AAAA: read udp 10.244.0.3:44330->169.254.169.254:53: i/o timeout
2019-05-27T11:20:13.222Z [ERROR] plugin/errors: 2 www.google.com. AAAA: read udp 10.244.0.3:58693->169.254.169.254:53: i/o timeout
2019-05-27T11:20:17.225Z [ERROR] plugin/errors: 2 www.google.com.google.internal. AAAA: read udp 10.244.0.3:38230->169.254.169.254:53: i/o timeout
2019-05-27T11:20:23.229Z [ERROR] plugin/errors: 2 www.google.com.google.internal. AAAA: read udp 10.244.0.3:55421->169.254.169.254:53: i/o timeout
2019-05-27T11:20:27.232Z [ERROR] plugin/errors: 2 www.google.com.c.eco-emissary-99515.internal. A: read udp 10.244.0.3:43721->169.254.169.254:53: i/o timeout
2019-05-27T11:20:29.233Z [ERROR] plugin/errors: 2 www.google.com.google.internal. A: read udp 10.244.0.3:50274->169.254.169.254:53: i/o timeout
2019-05-27T11:20:37.237Z [ERROR] plugin/errors: 2 www.google.com. A: read udp 10.244.0.3:42297->169.254.169.254:53: i/o timeout
2019-05-27T11:20:39.239Z [ERROR] plugin/errors: 2 www.google.com.c.eco-emissary-99515.internal. A: read udp 10.244.0.3:58562->169.254.169.254:53: i/o timeout
2019-05-27T11:20:41.241Z [ERROR] plugin/errors: 2 www.google.com.google.internal. A: read udp 10.244.0.3:38125->169.254.169.254:53: i/o timeout
2019-05-27T11:20:43.242Z [ERROR] plugin/errors: 2 www.google.com. A: read udp 10.244.0.3:47642->169.254.169.254:53: i/o timeout
2019-05-27T11:20:45.244Z [ERROR] plugin/errors: 2 www.google.com.c.eco-emissary-99515.internal. A: read udp 10.244.0.3:47169->169.254.169.254:53: i/o timeout
2019-05-27T11:20:47.245Z [ERROR] plugin/errors: 2 www.google.com.google.internal. A: read udp 10.244.0.3:51052->169.254.169.254:53: i/o timeout
.:53
```

--- a/pkg/build/node/const.go
+++ b/pkg/build/node/const.go
@@ -303,7 +303,7 @@ data:
   config: |-
     nonMasqueradeCIDRs:
       - {{ .PodSubnet }}
-    masqLinkLocal: false
+    masqLinkLocal: true
     resyncInterval: 60s
 ---
 apiVersion: extensions/v1beta1

